File: ShadowFrameworkLite20/src/shadow/system/data/objects/SFGenericFixedFloat.java
package shadow.system.data.objects;

import shadow.system.data.objects.SFBinaryValue;

public abstract class SFGenericFixedFloat extends SFBinaryValue {

	protected float multiplier ;
	protected float backmultiplier ;

	public abstract int getBitSize();
	public abstract SFGenericFixedFloat clone();

	public SFGenericFixedFloat(float multiplier, float backmultiplier) {
		super();
		this.multiplier = multiplier;
		this.backmultiplier = backmultiplier;
	}

	public float getFloat() {
		short value=(short)(this.value);
		return value*multiplier;
	}
	
	public void setFloat(float f) {
		short data=(short)(f*backmultiplier);
		this.value=(data<0?data+256*256:data);
	}

	@Override
	public void setValue(int value) {
		super.setValue(value);
	}

	@Override
	public void setStringValue(String value) {
		setFloat(new Float(value));
	}

	@Override
	public String toStringValue() {
		return getFloat()+"";
	}

}

File: trunk/ShadowFrameworkJS/src/shadow/system/data/objects/SFGenericFixedFloat.js


function SFGenericFixedFloat(multiplier,backmultiplier){
	this.multiplier = multiplier;
	this.backmultiplier = backmultiplier;
}

inherit(SFGenericFixedFloat,SFBinaryValue);


SFGenericFixedFloat.prototype["getFloat"]=function(){
			var COUNT=(1<<size);
			var MAX_POSITIVE=(1<<(size-1))-1;
			var value=this.value>=32768?(this.value-65536):this.value;
			return value*this.multiplier;
		};

SFGenericFixedFloat.prototype["setFloat"]=function(f){
			var data=Math.floor(f*this.backmultiplier);
			this.value=(data<0?data+256*256:data);
		};
